src/regs/mair_el1.rs

```diff
@@ -322,105 +322,105 @@ register_bitfields! {u64,
         ],
 
         /// Attribute 1
-        Attr1_HIGH OFFSET(12) NUMBITS(4) [
+        Attr1_Normal_Outer OFFSET(12) NUMBITS(4) [
             Device = 0b0000,
 
-            Memory_OuterWriteThrough_Transient_WriteAlloc = 0b0001,
-            Memory_OuterWriteThrough_Transient_ReadAlloc = 0b0010,
-            Memory_OuterWriteThrough_Transient_ReadAlloc_WriteAlloc = 0b0011,
+            WriteThrough_Transient_WAlloc = 0b0001,
+            WriteThrough_Transient_RAlloc = 0b0010,
+            WriteThrough_Transient_RWAlloc = 0b0011,
 
-            Memory_OuterNonCacheable = 0b0100,
-            Memory_OuterWriteBack_Transient_WriteAlloc = 0b0101,
-            Memory_OuterWriteBack_Transient_ReadAlloc = 0b0110,
-            Memory_OuterWriteBack_Transient_ReadAlloc_WriteAlloc = 0b0111,
+            NonCacheable = 0b0100,
+            WriteBack_Transient_WAlloc = 0b0101,
+            WriteBack_Transient_RAlloc = 0b0110,
+            WriteBack_Transient_RWAlloc = 0b0111,
 
-            Memory_OuterWriteThrough_NonTransient = 0b1000,
-            Memory_OuterWriteThrough_NonTransient_WriteAlloc = 0b1001,
-            Memory_OuterWriteThrough_NonTransient_ReadAlloc = 0b1010,
-            Memory_OuterWriteThrough_NonTransient_ReadAlloc_WriteAlloc = 0b1011,
+            WriteThrough_NonTransient = 0b1000,
+            WriteThrough_NonTransient_WAlloc = 0b1001,
+            WriteThrough_NonTransient_RAlloc = 0b1010,
+            WriteThrough_NonTransient_RWAlloc = 0b1011,
 
-            Memory_OuterWriteBack_NonTransient = 0b1100,
-            Memory_OuterWriteBack_NonTransient_WriteAlloc = 0b1101,
-            Memory_OuterWriteBack_NonTransient_ReadAlloc = 0b1110,
-            Memory_OuterWriteBack_NonTransient_ReadAlloc_WriteAlloc = 0b1111
+            WriteBack_NonTransient = 0b1100,
+            WriteBack_NonTransient_WAlloc = 0b1101,
+            WriteBack_NonTransient_RAlloc = 0b1110,
+            WriteBack_NonTransient_RWAlloc = 0b1111
```

Comment:
When I wrote mine (which I've pushed to my repo, btw), I liked:
`WriteBack_NonTransient_AllocRW`
rather than:
`WriteBack_NonTransient_RWAlloc`

Reply:
In the AArch64 ref manual they write it as `Write-Allocate policy` and `Read-Allocate policy`.
Therefore, I'd prefer to just stick to that ordering of words.

Also, I'd like to stick to the policy of this crate to be as elaborate as possible in the naming of constants (I didn't even follow through myself with the nGnRnE, for example). They are not programmed that often, so if the programming spans a few lines of source code, that's okay as long as it removes any last doubt of what's happening.

I like your ideas of having `Attr0_Device` and `Attr0_Normal_xx`, that makes it more concise!

Here's my proposal:

```rust
        Attr0_Device OFFSET(0) NUMBITS(8) [
            nonGathering_nonReordering_noEarlyWriteAck = 0b00000000,
            nGathering_nonReordering_EarlyWriteAck = 0b00000100,
            nonGathering_Reordering_EarlyWriteAck = 0b00001000,
            Gathering_Reordering_EarlyWriteAck = 0b00001100

...

        Attr0_Normal_Inner OFFSET(0) NUMBITS(4) [
            WriteThrough_Transient = 0x0000,
            WriteThrough_Transient_WriteAlloc = 0x0001,
            WriteThrough_Transient_ReadAlloc = 0x0010,
            WriteThrough_Transient_ReadWriteAlloc = 0x0011,
```

Reply:
About Device: I prefer the shorter notation (`nGnRnE` & friends), as it's used in other code-bases, and in the documentation (e.g. any use of `DeviceType` (J1-5398)). Then again, the allocation `MemHint`s are not really readable (J1-5397, `Allocation`). So I'm not really sure.

Reply:
Theoretically, we could have both.
Not sure if we should go down this path, though.